histopath/parsers/asap_parser.py

```diff
@@ -0,0 +1,88 @@
+"""ASAP format annotation parser."""
+
+import re
+import xml.etree.ElementTree as ET
+from pathlib import Path
+from typing import Iterable
+
+from shapely.geometry import Point, Polygon
+
+
+class ASAPParser:
+    """Parser for ASAP format annotation files.
+
+    ASAP (Automated Slide Analysis Platform) uses XML format for storing annotations.
+    This parser supports both polygon and point annotations.
+    """
+
+    def __init__(self, file_path: Path | str):
+        self.file_path = Path(file_path)
+        self.tree = ET.parse(self.file_path)
```

Comment:
```suggestion
    def __init__(self, file_path: Path | str):
        self.tree = ET.parse(file_path)
```
The type should be also more general. Possibly the same as `ET.parse` input type